projects/systelab-components/src/lib/contextmenu/abstract-context-menu.component.ts

```diff
@@ -61,15 +67,36 @@ export abstract class AbstractContextMenuComponent<T> extends AbstractContextCom
 
 	public doClick(event: any, elementID: string, action: ContextMenuOption, parent?: ContextMenuOption) {
 		if (this.isEnabled(elementID, action.actionId)) {
-			this.executeAction(event, elementID, action.actionId, parent ? parent.actionId : undefined);
+			this.executeAction(event, elementID, action.actionId);
+		}
+	}
+
+	public doClickWithString(event: any, elementID: string, actionId: string) {
```

Comment:
withString?

Reply:
Yes, because the action passed is not an action as in the click but an string from where we can extract the action.